src/m365/spfx/commands/project/project-upgrade/upgrade-1.12.0.ts

```diff
@@ -76,6 +77,7 @@ module.exports = [
   new FN002014_DEVDEP_types_es6_promise('', false),
   new FN002015_DEVDEP_types_react('16.9.36'),
   new FN002016_DEVDEP_types_react_dom('16.9.8'),
+  new FN002019_DEVDEP_spfx_fast_serve_helpers('~1.12.0'),
```

Comment:
The way DependencyRule is built right now, we can't pass a version range in here. Instead, it expects a specific version. If someone would install a specific version of spfx-fast-serve-helpers in their project, then we'd get a runtime exception on this line: https://github.com/pnp/cli-microsoft365/blob/c51892964f48e7c11ad3a7365af991c072d3e191/src/m365/spfx/commands/project/project-upgrade/rules/DependencyRule.ts#L58 because a range is not a valid version to compare using the semver's lt function. We could totally adjust our version comparison logic, but I wonder if for now it wouldn't be easier to pin a specific version here like we do with all other dependencies.

Reply:
I was somewhat doubtful myself whether this was possible or not.  During testing it always did install the latest patch version of the package. But here I forgot to check if the version control would include it. Currently I have taken out the tilde `~` and changed it to the first patch version for everything.